pkg/steps/multi_stage.go

```diff
@@ -551,11 +555,32 @@ func (s *multiStageTestStep) runPods(ctx context.Context, pods []coreapi.Pod, sh
 }
 
 func (s *multiStageTestStep) runPod(ctx context.Context, pod *coreapi.Pod, notifier *TestCaseNotifier) error {
+	// work around activeDeadlineSeconds vs. container status: https://github.com/kubernetes/kubernetes/issues/88611
+	activeDeadlineSeconds := pod.Spec.ActiveDeadlineSeconds
+	pod.Spec.ActiveDeadlineSeconds = nil
+	// ensure we notice before the pod is deleted
+	pod.ObjectMeta.Finalizers = []string{finalizer}
 	if _, err := createOrRestartPod(s.podClient.Pods(s.jobSpec.Namespace()), pod); err != nil {
 		return fmt.Errorf("failed to create or restart %q pod: %w", pod.Name, err)
 	}
+	podComplete := make(chan struct{}, 1)
+	if activeDeadlineSeconds != nil {
+		go func() {
+			duration := time.Duration(*activeDeadlineSeconds * 1000000000) // convert seconds to nanoseconds
+			select {
+			case <-podComplete:
+				// no need to delete
+			case <-time.After(duration):
+				log.Printf("cleanup: activeDeadlineSeconds %s exceeded, deleting pod %s", duration, pod.Name)
```

Comment:
nit: might be nicer for users with

```go
log.Printf("Pod %s did not finish before timeout, triggering deletion...", duration, pod.Name)
```

just so there's no jargon

Reply:
> ...might be nicer for users with...

I like name-dropping `activeDeadlineSeconds`, so folks know what to `grep` for in the config.  But I guess we're using `active_deadline_seconds` in the config as requested [here][1], so naming the Kube-side config property isn't as useful.  Pivoted to your suggestion with b97fd1cee -> 4e0590ace.

[1]: https://github.com/openshift/ci-tools/pull/1257#discussion_r494113890

Reply:
Obsolete with [the pivot to Vadim's approach][1].

[1]: https://github.com/openshift/ci-tools/pull/1257#issuecomment-705131035